src/ring_theory/roots_of_unity.lean

```diff
@@ -811,6 +811,62 @@ lemma minimal_polynomial_dvd_mod_p {p : ℕ} [hprime : fact p.prime] (hdiv : ¬
 (unique_factorization_monoid.dvd_pow_iff_dvd_of_squarefree (squarefree_minimal_polynomial_mod h
   hpos hdiv) (nat.prime.ne_zero hprime)).1 (minimal_polynomial_dvd_pow_mod h hpos hdiv)
 
+/-- If `p` is a prime that does not divide `n`, then the minimal polynomials of a root of unity `μ`
+and of `μ ^ p` are the same. -/
+lemma minimal_polynomial_eq_pow {p : ℕ} [hprime : fact p.prime] (hdiv : ¬ p ∣ n) :
+  minimal_polynomial (is_integral h hpos) =
+  minimal_polynomial (is_integral (pow_of_prime h hprime hdiv) hpos) :=
+begin
+  by_contra hdiff,
+  set P := minimal_polynomial (is_integral h hpos),
+  set Q := minimal_polynomial (is_integral (pow_of_prime h hprime hdiv) hpos),
+  have Pmonic : P.monic := minimal_polynomial.monic (is_integral h hpos),
+  have Qmonic : Q.monic := minimal_polynomial.monic (is_integral (pow_of_prime h hprime hdiv) hpos),
+  have Pirr : irreducible P := minimal_polynomial.irreducible (is_integral h hpos),
+  have Qirr : irreducible Q := minimal_polynomial.irreducible (is_integral
+    (pow_of_prime h hprime hdiv) hpos),
+  have PQprim : is_primitive (P * Q) := is_primitive.mul (monic.is_primitive Pmonic)
+    (monic.is_primitive Qmonic),
```

Comment:
```suggestion
  have PQprim : is_primitive (P * Q) := Pmonic.is_primitive.mul Qmonic.is_primitive,
```
I think this is shorter and more readable. You can do similar things in the rest of the proof.